client/www/scripts/modules/api-analytics/api.analytics.controllers.js

```diff
@@ -27,25 +29,40 @@ ApiAnalytics.controller('ApiAnalyticsController', [
       };
 
       $scope.getData = function(d, i, depth, initialModel){
+        $scope.showChartDataLoading = true;
         var def = $q.defer();
 
-        ApiAnalyticsService.getApiAnalyticsChartDataByNode(d, i, depth, $scope.server, initialModel)
-          .then(function(data){
-            var allData = {};
+        if (!$scope.isLoading) {
+          $scope.isLoading = true;
+          ApiAnalyticsService.getApiAnalyticsChartDataByNode(d, i, depth, $scope.server, initialModel)
+            .then(function(data){
+              var allData = {};
 
-            angular.extend(allData, data);
+              angular.extend(allData, data);
 
-            var chart = {
-              data: data,
-              node: d,
-              nodeIdx: i,
-              allData: allData
-            };
+              var chart = {
+                data: data,
+                node: d,
+                nodeIdx: i,
+                allData: allData
+              };
 
-            $scope.apiChart = chart;
+              $scope.apiChart = chart;
+              $scope.showChartDataLoading = false;
+              $scope.isLoading = false;
+              def.resolve($scope.apiChart);
+            })
+            .catch(function(error) {
+              $log.warn('bad get chart data: ' + JSON.stringify(error));
```

Comment:
i don’t think you need to stringify errors
when doing $log.warn do this:
`$log.warn(‘some message’, error);`
 instead of `$log.warn(‘some messsage’ + JSON.stringify(error));`